src/lumigo_tracer/auto_instrument_handler.py

```diff
@@ -1,13 +1,25 @@
 import os
+import sys
 
 from lumigo_tracer.lumigo_utils import is_aws_environment
 
 try:
     # Try to import AWS's current _get_handler logic
     from bootstrap import _get_handler as aws_get_handler
 except Exception:
-    # Import a snapshot of _get_handler from python38 runtime
-    from lumigo_tracer.libs.bootstrap import _get_handler as aws_get_handler
+    # Import a snapshot of _get_handler from the lambda runtime interface client (awslambdaric)
+    # matching the current python version
+    if sys.version_info >= (3, 7):
+        # This version supports python 3.7 up to 3.12
+        from lumigo_tracer.libs.awslambdaric.release_2_0_8.bootstrap import (
+            _get_handler as aws_get_handler,
+        )
+    else:
```

Comment:
are there cases that are supported in python 3.6 and not >3.7.?
Maybe the first import is sufficient?

In any case, I would remove it because you know for sure that in python3.6, you'll not get into this "except" - because the "try" will succeed for sure.
Initially, we created this fallback because we feared the unknown. Your case is really not unknown :)

Reply:
awslambdaric version 2.0.8 supports (officially) python 3.7 - 3.12 (And unofficially supports 3.6 according to the setup.py file)

awslambdaric version 2.0.0 supports 3.6 - 3.11.

So because the lib is used in a worst case scenario that awslamdbaric isn't available I think we should just use version 2.0.8